### PR TITLE
Adding CVE number when reporting a bug  

### PR DESCRIPTION
**Pull request by HanilJain**
### Added #1927 

While reporting bug now CVE number can be attached with the bug report. 
With an additional feature of validating check.

Screenshot 2024-03-07 040440

Screenshot 2024-03-07 040801

### CODE REVIEW

File: website/templates/issue2.html [hunk 1]
```diff
@@ -174,6 +174,10 @@
                                         <p class="text-3xl font-bold">Reported On:</p>
                                         <span class="font-bold bg-red-600 px-5 py-1 text-white rounded-xl">{{ object.created }}</span>
                                     </div>
+                                    <div class="w-full flex justify-between my-3">
+                                        <p class="text-3xl font-bold">CVE Number:</p>
```
Comment (arkid15r) [inline website/templates/issue2.html:178]: It's technically `CVE ID`

File: website/templates/report.html [hunk 2]
```diff
@@ -294,6 +295,19 @@ <h3 class="text-xl font-bold leading-none text-red-500">Latest Issues</h3>
                          class="w-[75%] ml-6 md:ml-0 flex items-end mt-6 flex-wrap min-h-[180px]"></div>
                 </div>
             </div>
+            <div>
+                <div class="cve_no">
+                    <label for="cve_no">CVE Number</label>
+                    <input class="mt-4  ml-2 w-100 rounded border-2"
+                           type="text"
+                           placeholder="CVE-XXXX-XXXX..."
+                           id="cve_no_input"
```
Comment (arkid15r) [inline website/templates/report.html:304]: Another reason to use better naming -- `no input` 🤷‍♂️

File: website/templates/report.html [hunk 3]
```diff
@@ -367,6 +381,24 @@ <h2 class="text-2xl font-semibold leading-7 text-gray-900">{% trans "ADD TEAM ME
 {% endblock content %}
 {% block after_js %}
     <script type="text/javascript">
+    function isCVEValid(cveNumber) {
+        const cveRegex = /^CVE-\d{4}-\d{4,}$/;
```
Comment (arkid15r) [inline website/templates/report.html:385]: It should be limited from the other side too
```suggestion
        const cveRegex = /^CVE-\d{4}-\d{4,7}$/;
```

File: website/templates/report.html [hunk 4]
```diff
@@ -367,6 +381,24 @@ <h2 class="text-2xl font-semibold leading-7 text-gray-900">{% trans "ADD TEAM ME
...
+        return cveRegex.test(cveNumber);
+    }
+
+    function validateCVE() {
+        const cveInput = document.getElementById('cve_no_input').value;
+        const validationResult = document.getElementById('cve_no_input');
+
+        if (isCVEValid(cveInput)) {
+            validationResult.style.borderColor = 'green';
+        }  else if (cveInput == None){
```
Comment (arkid15r) [inline website/templates/report.html:395]: What's `None` here?

**Review by arkid15r — COMMENTED** [review]
Hey @HanilJain good work!

A couple of suggestions to address:

**Comment by HanilJain** [comment]
> Hey @HanilJain good work!
> 
> A couple of suggestions to address:

@arkid15r Thanks for suggestion. I have made all the changes mentioned

File: website/templates/issue2.html (hunk 1 above)
Comment (HanilJain) [reply website/templates/issue2.html]: Ohkay 👍

File: website/templates/report.html (hunk 2 above)
Comment (HanilJain) [reply website/templates/report.html]: ahhh yess got it changes it to `cve_id_input`

File: website/templates/report.html (hunk 3 above)
Comment (HanilJain) [reply website/templates/report.html]: done 👍

**Review by DonnieBLT — APPROVED** [review]

File: website/models.py [hunk 5]
```diff
@@ -253,6 +253,7 @@ class Issue(models.Model):
     is_hidden = models.BooleanField(default=False)
     rewarded = models.PositiveIntegerField(default=0)  # money rewarded by the company
     reporter_ip_address = models.GenericIPAddressField(null=True, blank=True)
+    cve_id = models.CharField(max_length=160, null=True, blank=True)
```
Comment (arkid15r) [inline website/models.py:256]: I suggested making `max_length=16`

File: website/templates/report.html (hunk 4 above)
Comment (arkid15r) [reply website/templates/report.html:400]: This is JS context, not Python. It does not have `None`, right?

**Review by arkid15r — COMMENTED** [review]
Hey @DonnieBLT, it seems this PR has 2 unresolved issues. Didn't have a chance to finish my review 🤷‍♂️

**Comment by HanilJain** [comment]
@arkid15r I'll make these changes add in new PR

File: website/models.py (hunk 5 above)
Comment (HanilJain) [reply website/models.py:256]: Done in #1974

File: website/templates/report.html (hunk 4 above)
Comment (HanilJain) [reply website/templates/report.html:400]: I forget to remove it but have hade these changes in #1974

**Comment by HanilJain** [comment]
> Hey @DonnieBLT, it seems this PR has 2 unresolved issues. Didn't have a chance to finish my review 🤷‍♂️

@arkid15r Thank you for suggestion , I have made the changes in #1974

**Pull request merged**